getblocklab/block-lab
# Add the migration page

#### Changes
* The markup and CSS is from Rob's Gist 😄  
* Migrates Block Lab blocks to Genesis Custom Blocks
* Adds a 'Migrate' submenu under `/wp-admin > 'Block Lab'`

![migration-full](https://user-images.githubusercontent.com/4063887/87693998-3f833d80-c753-11ea-8d30-fe2d9fbc2a6c.gif)


#### Testing instructions (updated July 27th)
1. Upload and activate the plugin [block-lab.zip](https://github.com/getblocklab/block-lab/files/4983292/block-lab.zip), or do `composer install && npm install && npm run build`
2. If needed, [populate some Block Lab blocks](https://github.com/getblocklab/block-lab/pull/606#issuecomment-646743052)
3. Either install the GCB plugin: [genesis-custom-blocks.zip](https://github.com/getblocklab/block-lab/files/4983329/genesis-custom-blocks.zip), or do steps 5-7
4. Don't activate it yet
5. `cd path/to/wp-content/plugins/`
6. `git clone https://github.com/studiopress/genesis-custom-blocks` # if you haven't done this already
7. `cd genesis-custom-blocks && composer install && npm install && npm run build` # So that it's available to be activated in the final step
8. Go to `/wp-admin` > Block Lab > Migrate
9. Go through the migration flow 
10. Expected: The Block Lab blocks should now appear in Genesis Custom Blocks:
<img width="958" alt="bl-block-here" src="https://user-images.githubusercontent.com/4063887/87060407-1b15e700-c1d0-11ea-9478-44973b5d19e7.png">

11. Expected: Block Lab blocks in posts should look the same other than icons, and should now be Genesis Custom Blocks blocks:

<img width="1179" alt="icon-genesis-here" src="https://user-images.githubusercontent.com/4063887/87060518-3da80000-c1d0-11ea-9ab6-6bd29aa541e2.png">

12. Here's how to [reverse the migration](https://github.com/getblocklab/block-lab/pull/629#issuecomment-655846402)
